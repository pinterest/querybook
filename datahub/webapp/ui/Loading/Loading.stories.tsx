import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import centered from '@storybook/addon-centered/react';

import { Loading, LoadingIcon, LoadingRow } from './Loading';

export const _Loading = () => (
    <div>
        <Loading />
    </div>
);
export const _LoadingIcon = () => <LoadingIcon />;

export const _LoadingRow = () => <LoadingRow />;

export default {
    title: 'Stateless/Loading',
    decorators: [centered],
};
