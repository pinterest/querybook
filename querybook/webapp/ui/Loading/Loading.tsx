import React from 'react';
import styled from 'styled-components';

import { Icon } from 'ui/Icon/Icon';
import { Title } from 'ui/Title/Title';

export interface ILoadingProps {
    useSpinner?: boolean;
    text?: string;
    className?: string;

    fullHeight?: boolean;
    height?: number;
}

const StyledLoading = styled.div`
    position: relative;
    overflow: hidden;
    display: block;
    min-width: 100px;
    min-height: ${(props) => `${props.height ?? 240}px`};
    height: ${(props) => `${props.height ?? 360}px`};

    ${({ fullHeight }) =>
        fullHeight &&
        `
        height: 100%;
    `};

    > div {
        height: 100%;
    }
`;

const StyledLoadingText = styled.div`
    position: relative;
    text-align: center;
`;

export const LoadingIcon: React.FC<{ className?: string }> = ({
    className = '',
}) => (
    <div className="flex-center">
        <Icon name="Loading" className={'p8' + className} color="light" />
    </div>
);

export const LoadingRow: React.FC = () => (
    <div className="center-align mv4">
        <LoadingIcon />
    </div>
);

export const Loading: React.FunctionComponent<ILoadingProps> = ({
    useSpinner = true,
    text = '',
    className = '',
    height,
    fullHeight,
}) => {
    const loadingIcon = useSpinner ? <LoadingIcon /> : null;

    const loadingText =
        (text || '').length > 0 ? (
            <StyledLoadingText>
                <Title size="xxxlarge" color="light">
                    {text}
                </Title>
            </StyledLoadingText>
        ) : null;

    return (
        <StyledLoading
            className={'Loading flex-center ' + className}
            height={height}
            fullHeight={fullHeight}
        >
            {loadingIcon}
            {loadingText}
        </StyledLoading>
    );
};
