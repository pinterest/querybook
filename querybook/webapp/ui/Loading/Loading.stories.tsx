import centered from '@storybook/addon-centered/react';
import React from 'react';

import { Loading, LoadingIcon, LoadingRow } from './Loading';

export const _Loading = () => (
    <div>
        <Loading />
    </div>
);
export const _LoadingIcon = () => <LoadingIcon />;

export const _LoadingRow = () => <LoadingRow />;

export default {
    title: 'Stateless/Loading',
    decorators: [centered],
};
