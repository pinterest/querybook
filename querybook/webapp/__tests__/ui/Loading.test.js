import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';

import { Loading } from '../../ui/Loading/Loading';

it('renders without crashing', () => {
    shallow(<Loading />);
});

describe('matches enzyme snapshots', () => {
    it('matches snapshot', () => {
        const wrapper = shallow(<Loading />);
        const serialized = toJson(wrapper);
        expect(serialized).toMatchSnapshot();
    });
});

describe('matches test renderer snapshot', () => {
    it('serializes the styles', () => {
        const output = renderer.create(<Loading />);
        expect(output).toMatchSnapshot();
    });
});
