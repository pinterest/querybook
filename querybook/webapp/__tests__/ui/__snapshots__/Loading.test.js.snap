// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches enzyme snapshots matches snapshot 1`] = `
<Loading__StyledLoading
  className=""
>
  <LoadingIcon />
</Loading__StyledLoading>
`;

exports[`matches test renderer snapshot serializes the styles 1`] = `
<div
  className="Loading__StyledLoading-sc-1xendm7-0 cdKARu"
>
  <div
    className="Loading__StyledSpinner-sc-1xendm7-1 dUUxro"
  >
    <i
      className="fa fa-spinner fa-pulse"
    />
  </div>
</div>
`;
